syntax = "proto3";

service InterChatService {
    // 双向流 RPC：用于ChatServer 之间转发消息
    rpc ForwardMessage(stream ForwardedMessage) returns (stream ForwardedMessage);
}

message ForwardedMessage {
    string message_id = 1;
    string original_sender_id = 2;   // 发送者 uid
    string target_recipient_id = 3; // 接收者 uid
    string content = 4;              // 消息内容
    string timestamp = 5;            // 消息时间戳
}

service NotificationService {
    // 推送通知给指定用户
    rpc PushNotification(PushNotificationRequest) returns (PushNotificationResponse);
}

message PushNotificationRequest {
    int64 target_uid = 1;
    int32 notification_type = 2;
    string content_json = 3;
}

message PushNotificationResponse {
    bool success = 1;
}